### PR TITLE
Make sure unused places in ChannelHandleMap are initalized

### PR DESCRIPTION
**Pull request by daschuer**
The using code assumes that empty places are nullptr, which is not the case:

https://doc.qt.io/qt-5/qvarlengtharray.html
> QVarLengthArray doesn't initialize the memory if the value is a basic type. (QVector always does.)

This PR fixes the assumption. And hopefully the rasher in https://bugs.launchpad.net/mixxx/+bug/1775497 

Unfortunately the over all design is still broken and instead of crashing it will probably now hit VERIFY_AND_DEBUG_ASSERTs that have been added as safety net. 

I think for 2.4 a bigger refactoring is required with a better memory management, fixing these edge cases. 
Before building on it in #2618

### CODE REVIEW

**Review by Holzhaus — COMMENTED** [review]
Can't really comment on the changes. They build and I noticed no issues when testing. Let's wait a little until the original reporter had the chance to confirm that this fixes the crash.

File: src/engine/channelhandle.h [hunk 1]
```diff
@@ -160,6 +159,10 @@ class ChannelHandleMap {
     typedef typename QVarLengthArray<T, kMaxExpectedGroups>::const_iterator const_iterator;
     typedef typename QVarLengthArray<T, kMaxExpectedGroups>::iterator iterator;
 
+    ChannelHandleMap()
+            : m_dummy(T()) {
```
Comment (uklotzde) [inline src/engine/channelhandle.h:163]: I suggest to replace `T()` with `T{}` (here and when appending) to improve readability and to indicate that this is a constructor initialization.

File: src/engine/channelhandle.h [hunk 2]
```diff
@@ -208,12 +211,18 @@ class ChannelHandleMap {
...
-        if (m_data.size() < iSize) {
-            m_data.resize(iSize);
+        if (QTypeInfo<T>::isComplex) {
+            // The value for complex types is initialized by QVarLengthArray
+            if (m_data.size() < iSize) {
+                m_data.resize(iSize);
+            }
+        } else {
+            // We need to initialize simple types ourselves
+            while (m_data.size() < iSize) {
```
[the remark this answers is not in the document]
Comment (daschuer) [reply src/engine/channelhandle.h:221]: We initial reserve kMaxExpectedGroups = 256 elements to avoid allocations.

**Comment by daschuer** [comment]
Done

**Review by uklotzde — APPROVED** [review]
Thank you! No risk to break anything. Let's hope the bug is fixed. LGTM

**Pull request merged**